docs/advanced/integration_tests.md

```diff
@@ -0,0 +1,130 @@
+```go
+func (s *ApiTestSuite) Test_Euro() *suite.ApiServerTestCase {
+	return &suite.ApiServerTestCase{
+		Method:             http.MethodGet,
+		Url:                "/euro/10/GBP",
+		Headers:            map[string]string{},
+		ExpectedStatusCode: http.StatusOK,
+		Assert: func(response *resty.Response) error {
+			s.Equal(http.StatusOK, response.StatusCode())
+
+			result, err := strconv.ParseFloat(string(response.Body()), 64)
+			s.NoError(err)
+			s.Equal(8.0, result)
+
+			return nil
+		},
+	}
+}
+```
+
+Notice the predefined structure of an _ApiServerTestCase_: it is an object that has fields for all the information needed in performing an http call, including a method that tests the result for correctness.
+
+### Stream integration tests
+
+If you have a Gosoline application that takes its input from a stream, you need a test which can run your application locally, send data to a stream, and check your application's output for correctness. 
+
+The `stream-consumer` application, found in `advanced/stream-consumer`, reads unsigned integers from an `inputEvent` sqs queue, increments them by one and publishes them to the `publisher-outputEvent` SNS topic. Both the SQS queue which provides it with input, and the SNS topic to which it publishes its results, are configured in its `config.dist.yml` file. Below is an extract from its integration test:
+
+[embedmd]:# (../../examples/advanced/stream-consumer-test/stream_consumer_test.go /func \(s \*ConsumerTestSuite\) SetupSuite/ /\n}/)
+```go
+func (s *ConsumerTestSuite) SetupSuite() []suite.Option {
+	return []suite.Option{
+		suite.WithLogLevel("debug"),
+		suite.WithConfigFile("../stream-consumer/config.dist.yml"),
+		suite.WithModule("consumer", stream.NewConsumer("consumer", consumer.NewConsumer())),
+	}
+}
+```
+
+It is making use of the same `config.dist.yml` file as `stream-consumer`, and it will be using the module created by _consumer.NewConsumer_. The difference between a _StreamTestCase_ test and an _ApiServerTestCase_ can be seen in its test cases:
+
+[embedmd]:# (../../examples/advanced/stream-consumer-test/stream_consumer_test.go /func \(s \*ConsumerTestSuite\) TestSuccess/ /\n}/)
+```go
+func (s *ConsumerTestSuite) TestSuccess() *suite.StreamTestCase {
+	input := mdl.Uint(5)
+
+	return &suite.StreamTestCase{
+		Input: map[string][]suite.StreamTestCaseInput{
+			"consumer": {
+				{
+					Attributes: nil,
+					Body:       &input,
+				},
+			},
+		},
+		Assert: func() error {
+			result := mdl.Uint(0)
+			s.Env().StreamOutput("publisher-outputEvent").Unmarshal(0, result)
+
+			s.Equal(mdl.Uint(6), result)
+
+			return nil
+		},
+	}
+}
+```
+
+Similarly to an _ApiServerTestCase_, this _StreamTestCase_ is an object defining an input and an `Assert` function. Gosoline will run the module, send this input into its SNS queue, then run `Assert`. The `Assert` function reads the first element from a stream, _publisher-outputEvent_, then compare it with an expected result.
```

Comment:
What is an SNS queue? An integration test doesn't access any AWS services, so speaking of them is most likely incorrect